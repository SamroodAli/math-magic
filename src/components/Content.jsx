import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import QuoteAuthor from './QuoteImage';
import NextButton from './NextButton';

const Content = (
  {
    quote,
    authorImg,
    author,
    getQuotes,
  },
) => (
  <Container>
    <QuoteAuthor authorImg={authorImg} author={author} />
    <NextButton onClick={getQuotes} />
    <h3 data-testid="quote">{quote}</h3>
    <h4 data-testid="author">{author}</h4>
  </Container>
);

Content.propTypes = {
  quote: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  authorImg: PropTypes.string.isRequired,
  getQuotes: PropTypes.func.isRequired,
};

export default Content;
