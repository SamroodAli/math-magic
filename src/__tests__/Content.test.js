import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';
import Content from '../components/Content';

test('Content snapshot testing', () => {
  const tree = renderer.create(<Content quote="lorem ipsum" author="lorem" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Loader presence testing', async () => {
  const content = render(<Content quote="Loading" author="" />);
  const loader = await content.findByTestId('loader');
  expect(loader).toBeDefined();
});

test('Quote presence testing', async () => {
  const quote = 'Lorem ipsum';
  const author = 'Lorem';
  const content = render(<Content quote={quote} author={author} />);
  const quoteJsx = await content.findByTestId('quote');
  expect(quoteJsx.innerHTML).toEqual(quote);
});

test('Author presence testing', async () => {
  const quote = 'Lorem ipsum';
  const author = 'Lorem';
  const content = render(<Content quote={quote} author={author} />);
  const authorJsx = await content.findByTestId('author');
  expect(authorJsx.innerHTML).toEqual(author);
});
