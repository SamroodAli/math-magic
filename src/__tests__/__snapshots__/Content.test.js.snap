// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content snapshot testing 1`] = `
Array [
  <div
    style={
      Object {
        "height": "400px",
      }
    }
  >
    <img
      alt="lorem"
      className="ui circular centered image"
      height="400px"
    />
  </div>,
  <button
    className="ui black animated button"
    onClick={[Function]}
    style={
      Object {
        "margin": "1rem",
      }
    }
  >
    <div
      className="visible content"
    >
      Next
    </div>
    <div
      className="hidden content"
    >
      <i
        aria-hidden="true"
        className="arrow right icon"
        onClick={[Function]}
      />
    </div>
  </button>,
  <h3
    data-testid="quote"
  >
    lorem ipsum
  </h3>,
  <h4
    data-testid="author"
  >
    lorem
  </h4>,
]
`;
